src/py42/clients/cases.py

```diff
@@ -0,0 +1,196 @@
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 100.
+            sort_direction (str, optional): The direction on which to sort the response,
+                based on the corresponding sort key. `asc` or `desc`. Defaults to `asc`.
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number".
+                Available options are `name`, `number`, `createdAt`, `updatedAt`, `status`, `assigneeUsername`, `subjectUsername`.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of cases.
+        """
+        if status and status.upper() not in CaseStatus.choices():
+            raise Py42InvalidCaseStatus(status)
+
+        created_at = CasesClient._make_range(min_create_time, max_create_time)
+        updated_at = CasesClient._make_range(min_update_time, max_update_time)
+
+        return self._cases_service.get_all(
+            name=name,
+            status=status,
+            created_at=created_at,
+            updated_at=updated_at,
+            subject=subject,
+            assignee=assignee,
+            page_number=page_number,
+            page_size=page_size,
+            sort_direction=sort_direction,
+            sort_key=sort_key,
+            **kwargs
+        )
+
+    def get_case(self, case_number):
+        """Retrieve case details by case number.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        return self._cases_service.get_case(case_number)
+
+    def export_summary(self, case_number):
+        """Provides case summary to download as a PDF file.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        return self._cases_service.export_summary(case_number)
+
+    def update(
+        self,
+        case_number,
+        name=None,
+        subject=None,
+        assignee=None,
+        description=None,
+        findings=None,
+        status=None,
+    ):
+        """Updates case details for the given case number.
+
+        Args:
+            case_number (int): Case number of the case.
+            name (str, optional): Name of the case. Defaults to empty string.
+            subject (str, optional): A subject of the case. Defaults to empty string.
+            assignee (str, optional): User UID of the assignee. Defaults to empty string.
+            description (str, optional): Description of the case. Defaults to empty string.
+            findings (str, optional): Notes on the case. Defaults to empty string.
+            status (enum, optional): Status of the case. Defaults to None.
```

Comment:
The `Defaults to None.` should be at the end of the doc str.

Reply:
Moved.